spec/models/claim/matching_attribute_finder_spec.rb

```diff
@@ -0,0 +1,83 @@
+require "rails_helper"
+
+RSpec.describe Claim::MatchingAttributeFinder do
+  describe "#claim_ids_with_matching_attributes" do
+    let(:source_claim) {
+      create(
+        :claim,
+        teacher_reference_number: "0902344",
+        national_insurance_number: "QQ891011C",
+        email_address: "genghis.khan@mongol-empire.com",
+        bank_account_number: "34682151",
+        bank_sort_code: "972654",
+        building_society_roll_number: "123456789/ABCD",
+      )
+    }
+
+    let!(:claim_with_no_matching_attributes) { create(:claim, :submitted) }
+    let!(:unsubmitted_claim_with_matching_teacher_reference_number) { create(:claim, :submittable, teacher_reference_number: source_claim.teacher_reference_number) }
+
+    let(:claim_with_matching_attribute) { build(:claim, :submitted) }
```

Comment:
I got a bit confused when I read this `let` because, as it stands, it is _not_ a claim with a matching attribute; it only becomes one when you modify it inside the `it` blocks. Perhaps it would be clearer to create this claim each time you need it, inside the `it`?